### PR TITLE
Feat/iteration 3 network requests/async js

### PR DESCRIPTION
**Pull request by mikeharrison57**
**[Project Ticket](https://github.com/mikeharrison57/rancid-tomatillos/projects/1#card-83976391)**

**Describe the changes made to the codebase:**

- Refactored App class component to use network requests to fetch data from API's.
- Implemented error handling in `componentDidMount()` and `handleClick()`.  
- Refactored App state to include error property. 
- Installed cypress and wrote some initial tests.

**How was it tested?**

- Tested error rendering by deleting portion of API endpoint. Error status text does render when error occurs. 
- `console.log()` data from API endpoints.
- Wrote cypress test for errors. It works conditionally when errors occur, but not based on faulty server provided.
- Will need to address this test in future iteration. 

**Type of Change**

- [ ] Bug Fix
- [X] New Feature

**Issues**

- [Issue Not Fully Closed](https://github.com/mikeharrison57/rancid-tomatillos/projects/1#card-83976455)

**Checklist**

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] My changes generate no new bugs

### CODE REVIEW

**Review by nhodnett — COMMENTED** [review]
Good work on Iteration 3. We'll get the cypress tests working tomorrow after the Cypress lesson no doubt!

File: cypress/e2e/spec.cy.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+describe('main page', () => {
+  beforeEach(() => {
+    cy.visit('http://localhost:3000/')
+    cy.intercept('GET', 'https://rancid-tomatillos.herokuapp.com/api/v2/movies', {fixture: 'movies'})
```
Comment (Kalikoze) [inline cypress/e2e/spec.cy.js:4]: Great job setting up your Cypress tests already and utilizing fixtures!  🎉

File: src/App.js [hunk 2]
```diff
@@ -1,57 +1,67 @@
...
+      .then(res => {
+        if (res.ok) {
+          return res.json()
+        }
+        throw Error(res.statusText)
+        })
+      .then(data => this.setState({movies: data.movies}))
+      .catch(error => {
+        this.setState({error: error.message})
+      })
```
Comment (Kalikoze) [inline src/App.js:29]: Nice error handling here for handling 400 and 500 responses.  A small refactor you could do here is move your network requests into its own `apiCalls.js` file.

File: src/App.js [hunk 3]
```diff
@@ -1,57 +1,67 @@
...
-    console.log(this.state.movies)
-    console.log(movie.movie)
+    console.log(this.state.error)
     return (
       <div>
         <Navbar />
-        {this.state.movieSelected ?
+        {this.state.error ?
+        <h1 className='Error'>{this.state.error}</h1> 
+        :
```
Comment (Kalikoze) [inline src/App.js:53]: I'm a little confused by the nested ternary logic.  I might refactor this into something like:

```
{ this.state.error && <h1 className='Error'>{this.state.error}</h1>}
{ this.state.movieSelected && <IndividualMovie /> }
```

File: src/IndividualMovie.js [hunk 4]
```diff
@@ -9,7 +9,7 @@ const IndividualMovie = (props) => {
     return(
         <div className='IndividualMovie'>
             {/* <p>{id}</p> */}
-            <p>{title}</p>
+            <p className='Title'>{title}</p>
```
Comment (Kalikoze) [inline src/IndividualMovie.js:12]: I noticed you were using capitalized classNames.  I would still stick with kebab-case for classes.

**Review by Kalikoze — COMMENTED** [review]
Nice work on your PR!  I love that you've already started testing with Cypress, great job!  I made a couple of suggestions including moving your network requests into their own separate apiCalls file and refactoring the nested ternary logic, but otherwise this is looking great! 🎉

**Pull request merged**